core/src/main/scala/org/http4s/rho/bits/Metadata.scala

```diff
@@ -25,4 +25,4 @@ trait SecurityScopesMetaData extends Metadata {
 case class QueryMetaData[F[_], T](name: String, description: Option[String], p: QueryParser[F, T], default: Option[T], m: TypeTag[T]) extends Metadata
 
 /** Metadata about a header rule */
-case class HeaderMetaData[T <: HeaderKey.Extractable](key: T, default: Boolean) extends Metadata
+case class HeaderMetaData[T <: HeaderKey.Extractable](key: T, isRequired: Boolean) extends Metadata
```

Comment:
It ate my comment...

This could lead to boolean blindness vs. something like `Required | Optional`.  But your approach has the virtue of being less breaking.  I'd probably leave it like this, but just a thought.

Reply:
Hopefully people don't use the metadata directly often if ever.  Do you know of some ADT/Enum with `Required/Optional` that i can plugin here?  I don't really feel like it would be worth the effort (at this time anyways) to add them if there isn't some ready made solution already a dependency.

Reply:
No, nothing standard.  Roll with what's here, I'd say.